app/src/main/java/ch/epfl/skysync/database/FirestoreDatabase.kt

```diff
@@ -62,6 +62,40 @@ class FirestoreDatabase(private val useEmulator: Boolean = false) {
         }
   }
 
+  /**
+   * Set a new item to the database
+   *
+   * Set item at id (if null a new id is generated) and override any previously set id.
+   *
+   * @param path A filesystem-like path that specify the location of the table
+   * @param item The item to set to the database, the types of the attributes have to be Firestore
+   * @param id the id of the item (if null a new id is generated)
+   * @param onCompletion Callback called on completion of the operation
+   * @param onError Callback called when an error occurs
+   */
+  fun <T : Any> setItem(
+      path: String,
+      id: String?,
+      item: T,
+      onCompletion: (id: String) -> Unit,
+      onError: (Exception) -> Unit
+  ) {
+    if (id != null) {
+      db.collection(path)
+          .add(item)
```

Comment:
Pretty sure it should not be `add` (in your whole method you never use the `id` argument so there definitely something fishy...)

Reply:
Indeed this should be .document(id).set(item)